### PR TITLE
Orochi Network proposal for orosign

### PR DESCRIPTION
**Pull request by chiro-hiro**
#### Project Abstract

Orosign is a self-managing custodial Mobile App base on MPC and ECDSA to provide high security in digital assets/identity management that suits all demands of Web3.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@chiro-hiro:matrix.org`

### CODE REVIEW

File: applications/orochi-network-orosign.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+### Milestone 1
+
+- **Estimated Duration:** 1 month
+- **FTE:** 1.6
+- **Costs:** $5,600
+
+| Number  | Deliverable | Specification                                                                                                                       |
+| ------- | ----------- | ----------------------------------------------------------------------------------------------------------------------------------- |
+| **0a.** | License     | CC0 1.0                                                                                                                             |
+| **0b.** | Research    | Researching about Polkadot signature system                                                                                         |
```
Comment (keeganquigley) [inline applications/orochi-network-orosign.md:97]: Can you elaborate on what you mean by this?

**Comment by keeganquigley** [comment]
Thanks for the application @chiro-hiro in addition to the questions above:

- Is this meant to be an enterprise-grade security wallet such as this [SaaS wallet](https://www.atato.com/)?
- Will it be cross-chain compatible or is it only meant for substrate chains?
- Can you further expand on the technical details of the wallet in the deliverables? Thanks!

File: applications/orochi-network-orosign.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+| **0c.** | Testing and Testing Guide            | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests.                                                                       |
+| **0d.** | Docker                               | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone.                                                                                                                         |
+| **1.**  | Functionality: Polkadot in back-end  | Support Polkadot ecosystem in back-end of Orosign                                                                                                                                                                                     |
+| **2.**  | Functionality: Polkadot in front-end | Support Polkadot ecosystem in front-end of Orosign                                                                                                                                                                                    |
```
[the remark this answers is not in the document]
Comment (chiro-hiro) [reply applications/orochi-network-orosign.md:117]: @keeganquigley We write a service in Node.js to parsing data from Polkadot blockchain and its parachains. Front-end side we using React Native.

File: applications/orochi-network-orosign.md (hunk 1 above)
Comment (chiro-hiro) [reply applications/orochi-network-orosign.md:97]: We want to be fully compatible with Polkadot that meant we need to support all ECDSA that Polkadot support and see how could we apply MPC threshold signature for all of them.

**Comment by chiro-hiro** [comment]
Thank @keeganquigley, here are our answers.

> Is this meant to be an enterprise-grade security wallet such as this [SaaS wallet](https://www.atato.com/)?

We're self-custodian wallet, we offer enterprise-grade security for semi-retail users.

> Will it be cross-chain compatible or is it only meant for substrate chains?

In the budget of this proposal we only able to support Polkadot's chain and its parachains.

> Can you further expand on the technical details of the wallet in the deliverables? Thanks!

- We implement DKG to generate secret shares that allow participants to perform signing process without actual private key.
- User will hold let's say 3 shares (2 sign shares and 1 backup) of 5 shares (Orochi Network hold 2). We created a threshold signature 3 of 5, it's required at least 1 share from user to perform transaction co-signing.
    - Orochi Network can't perform the signing process.
    - User can perform the signing process with or without Orochi Network.

**Comment by keeganquigley** [comment]
Thanks for your reply @chiro-hiro however I'm not seeing that any changes have been made to the delivery tables. Can you update these to include the above information? 

- The mandatory deliverables look okay (0a. - 0d.) but the rest could be fleshed out a bit more.
- Milestone 1 research could also be expanded upon. What is the knowledge that will ultimately be generated by this research?
- What kind of report(s) will be produced?
- How are some of the questions you are hoping to find answers to during your research? Are there some common problems/issues with MPC wallets that the whole community would benefit from solving?

**Comment by chiro-hiro** [comment]
> Thanks for your reply @chiro-hiro however I'm not seeing that any changes have been made to the delivery tables. Can you update these to include the above information?

I've updated these information in the following commit.

> * The mandatory deliverables look okay (0a. - 0d.) but the rest could be fleshed out a bit more.

I added more details, please let me know if it is enough or not

> * Milestone 1 research could also be expanded upon. What is the knowledge that will ultimately be generated by this research?

The research is necessary to secure the success of the project, it provide the background knowledge in academic side to build up MPC wallet and also consider the technical trade-off between different MPC protocol and existing solution to perform MPC.

1. The ECDSA is fully compatible with MPC (each curve might have different scheme and parameters that need different approaches to keep the security level won't be drop below the standard).
2. Implement a MPC threshold signature won't create any issue that could break MPC Wallet security that allows any adversary to compromise the security or extracting private key from the MPC process. 
3. Provide the document that help to summary the implementation and deep understanding of the project.

> * What kind of report(s) will be produced?

We used to write a technical report about what we do an record the technical decision of the project.

Screenshot 2023-05-25 at 14 21 16

> * How are some of the questions you are hoping to find answers to during your research? Are there some common problems/issues with MPC wallets that the whole community would benefit from solving?

We want to find a good MPC model to perform ECDSA or build up a flexible MPC Wallet that can adjust the level of trade-off between the communication and computation. I think the common problem that every MPC is facing is communication cost, we try to do a trade-off with computation, we can build up a mechanism to balance them. Other project can use our research result to improve their MPC Wallet in the future and build better and securer MPC Wallet. I hope we can contribute to make Web3 more secure than the current situation.

**Comment by keeganquigley** [comment]
Thanks @chiro-hiro for your answers and for the changes. I will go ahead and mark the application as ready for review and ping the committee.

Personally, I won't approve it yet because I think 6 milestones is a bit ambitious, and I think you could break it down further into two grants. For example an initial level 2 grant focusing on the research and a PoC of the curve implementations, to be tailed by a follow-up grant for the back-end and front-end of the wallet. But perhaps others in the committee will feel differently.

**Comment by chiro-hiro** [comment]
Thank @keeganquigley for your valuable feedback , your comment make sense to me. I will split it into two level 2 proposals. One focus on R&D and PoC, another focus on implementation.

**Comment by chiro-hiro** [comment]
I'm close this pull request and move the first part of proposal to #1786

**Pull request closed without merging**